src/librustdoc/html/static/main.js

```diff
@@ -2952,6 +2937,35 @@ function defocusSearchBar() {
 
     onHashChange(null);
     window.onhashchange = onHashChange;
+
+    function loadScript(url) {
+        const script = document.createElement('script');
+        script.src = url;
+        document.head.append(script);
+    }
+
+    window.searchLoaded = false;
+    function loadSearch() {
+        if (!window.searchLoaded) {
+            window.searchLoaded = true;
+            loadScript(window.searchJS);
+        }
+    }
+
+    // Assumes `crates{version}.js` has already been loaded, setting the global const `crates`.
+    addSearchOptions(crates);
+    addSidebarCrates(crates);
+    var crateSearchDropDown = document.getElementById("crate-search");
```

Comment:
Can you put all this in a function too so we can avoid creating variables in the "global" scope of this function please?